src/urlencoded.rs

```diff
@@ -1,11 +1,137 @@
-use std::collections::HashMap;
-use url::form_urlencoded;
 use groupable::Groupable;
+use hyper::uri::RequestUri;
+use hyper::uri::RequestUri::{Star, AbsoluteUri, AbsolutePath, Authority};
+use plugin::{Plugin, Pluggable};
+use request::Request;
+use status::StatusCode;
+use std::collections::HashMap;
+use std::io::{Error, Read};
+use typemap::Key;
+use url::{form_urlencoded, UrlParser};
+
+type QueryStore = HashMap<String, Vec<String>>;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Query(QueryStore);
```

Comment:
If this is going to get reused for both of these, then we should probably rename it. I think we'll have to leave a re-export for `pub type Query = NewTypeName` incase downstream code is depending on this typename.

Perhaps `MultiMap`?